File: 169-majority-element/majority-element.py


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        counts = defaultdict(int)
        n = len(nums) // 2

        for num in nums:
            counts[num] += 1

        for num, count in counts.items():
            if count > n:
                return num

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int candidate, count = 0;

        for (int i = 0; i < n; i++) {
            if (count == 0) {
                count = 1;
                candidate = nums[i];
            } else {
                if (candidate == nums[i]) {
                    count++;

                } 
                else
                {
                    count--;
                }
            }
        }
        return candidate;
    }
};